internal/core/labels_test.go

```diff
@@ -0,0 +1,37 @@
+package core
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMergeCustomLabels(t *testing.T) {
+	t.Run("merge success", func(t *testing.T) {
```

Comment:
suggestion: keep test names simple and avoid spaces to they can be search for, spaces are placed by underscore.
```suggestion
	t.Run("success", func(t *testing.T) {
```